strapi/strapi
# [Data transfer] Handle signals to gracefully interrupt an ongoing transfer

### What does it do?

- Transfer engine provides an `abortTransfer()` method to interrupt an ongoing transfer

### Why is it needed?

- To provide a general programmatic way for transfers to be interrupted
- So that signals  SIGINT (Ctrl+C), SIGQUIT, and SIGTERM can be caught and gracefully handled to close a transfer/import/export using that mechanism
- it fixes a poorly implemented temporary solution that was added in 4.6

### How to test it?

Run a `strapi transfer` (or import/export) command, and press CTRL+C and the shutdown should be handled properly; the connection closed and the console messages updated appropriately to describe what happened.

Alternatively, use a scripted transfer with a timer to call engine.abort() and ensure that it aborts properly.

I don't actually have any good ideas for how to write automated tests for this given the nature of it, so that will come later once we have a way of running integration tests with the CLI.
